README.md

```diff
@@ -2,4 +2,207 @@
 
 [![codecov](https://codecov.io/github/opencomputeproject/ocp-diag-core-rust/graph/badge.svg?token=IJOG0T8XZ3)](https://codecov.io/github/opencomputeproject/ocp-diag-core-rust)
 
-WIP
\ No newline at end of file
+The **OCP Test & Validation Initiative** is a collaboration between datacenter hyperscalers having the goal of standardizing aspects of the hardware validation/diagnosis space, along with providing necessary tooling to enable both diagnostic developers and executors to leverage these interfaces.
+
+Specifically, the [ocp-diag-core-rust](https://github.com/opencomputeproject/ocp-diag-core-rust) project makes it easy for developers to use the **OCP Test & Validation specification** artifacts by presenting a pure-rust api that can be used to output spec compliant JSON messages.
+
+To start, please see below for [installation instructions](https://github.com/opencomputeproject/ocp-diag-core-rust#installation) and [usage](https://github.com/opencomputeproject/ocp-diag-core-rust#usage).
+
+This project is part of [ocp-diag-core](https://github.com/opencomputeproject/ocp-diag-core) and exists under the same [MIT License Agreement](https://github.com/opencomputeproject/ocp-diag-core-rust/LICENSE).
+
+### Installation
+
+Stable releases of the **ocp-diag-core-rust** codebase are published to **crates.io** under the package name [ocptv](https://crates.io/crates/ocptv), and can be easily installed with rust package manager.
+
+For general usage, the following steps should be sufficient to get the latest stable version using the [Package Installer for Rust](https://github.com/rust-lang/cargo):
+
+- **\[option 1]** using `cargo add`
+
+    ```bash
+    $ cargo add ocptv
+    ```
+
+- **\[option 2]** specifying it in Cargo.toml file
+
+    
+    ```toml
+    [dependencies]
+    ocptv = "1.0.0"
+    ```
+    and then run
+    ```bash
+    $ cargo update
+    ```
+
+To use the bleeding edge instead of the stable version, the git repository should be cloned.
+This assumes that the clone is manually kept up to date by git pulling whenever there are new commits upstream. All of the installation methods below will automatically use the latest library code.
+
+First clone the upstream latest code:
+```bash
+$ git clone https://github.com/opencomputeproject/ocp-diag-core-rust.git
+$ cd ocp-diag-core-rust
+$ git checkout dev # dev branch has the latest code
+```
+Then edit Cargo.toml in your project and add
+
+```
+[dependencies]
+ocptv = { version = "1.0.0", path = "/path/to/ocp-diag-core-rust" }
```

Comment:
no need to clone; the version number is only necessary when the user would try to publish their own artifact, it's not needed for dev work

```suggestion
ocptv = { git = "https://github.com/opencomputeproject/ocp-diag-core-rust.git", branch = "dev" }
```